app/services/proforma_service/convert_exercise_to_task.rb

```diff
@@ -77,62 +77,85 @@ def uuid
-          ms_file.usage_by_lms = 'display'
-        end,
-      ]
+      task_file(file).tap do |ms_file|
+        ms_file.used_by_grader = false
+        ms_file.usage_by_lms = 'display'
+      end
     end
 
     def tests
-      @exercise.files.filter(&:teacher_defined_assessment?).map do |file|
+      @exercise.files.filter(&:teacher_defined_assessment?).group_by {|file| xml_id_from_file(file).first }.map do |xml_id, files|
         ProformaXML::Test.new(
-          id: file.id,
-          title: file.name,
-          files: test_file(file),
-          meta_data: test_meta_data(file)
+          id: xml_id || files.first.id,
+          title: files.first.name,
+          files: files.map {|file| test_file(file) },
+          meta_data: test_meta_data(files)
         )
       end
     end
 
-    def test_meta_data(file)
+    def xml_id_from_file(file)
+      type = if file.teacher_defined_assessment?
+               'test'
+             elsif file.reference_implementation?
+               'ms'
+             else
+               'file'
+             end
+      xml_id_path_parts = file.xml_id_path&.split('/')
+      xml_id_path  = []
+
+      xml_id_path << (xml_id_path_parts&.first || "co-#{type}-#{file.id}") unless type == 'file'
+      xml_id_path << (xml_id_path_parts&.last || file.id)
+
+      xml_id_path
+    end
+
+    def test_meta_data(files)
       {
         '@@order' => %w[test-meta-data],
         'test-meta-data' => {
-          '@@order' => %w[CodeOcean:feedback-message CodeOcean:weight CodeOcean:hidden-feedback],
+          '@@order' => %w[CodeOcean:test-file],
           '@xmlns' => {'CodeOcean' => 'codeocean.openhpi.de'},
-          'CodeOcean:feedback-message' => {
-            '@@order' => %w[$1],
-            '$1' => file.feedback_message,
-          },
-          'CodeOcean:weight' => {
-            '@@order' => %w[$1],
-            '$1' => file.weight,
-          },
-          'CodeOcean:hidden-feedback' => {
-            '@@order' => %w[$1],
-            '$1' => file.hidden_feedback,
-          },
+          'CodeOcean:test-file' => files.to_h do |file|
+            [
+              "CodeOcean:#{xml_id_from_file(file).last}", {
+                '@@order' => %w[CodeOcean:feedback-message CodeOcean:weight CodeOcean:hidden-feedback],
+                '@xmlns' => {'CodeOcean' => 'codeocean.openhpi.de'},
+                '@id' =>  xml_id_from_file(file).last,
+                '@name' => file.name,
+                'CodeOcean:feedback-message' => {
+                  '@@order' => %w[$1],
+                  '$1' => file.feedback_message,
+                },
+                'CodeOcean:weight' => {
+                  '@@order' => %w[$1],
+                  '$1' => file.weight,
+                },
+                'CodeOcean:hidden-feedback' => {
+                  '@@order' => %w[$1],
+                  '$1' => file.hidden_feedback,
+                },
+              }
+            ]
+          end,
```

Comment:
I have the feeling that something is odd with the meta data. In my tests, I got the following results with openHPI/CodeHarbor@da1ecdda:

```xml
<test-meta-data xmlns:CodeOcean="codeocean.openhpi.de">
  <CodeOcean:test-file/>
</test-meta-data>
```

(despite having regular data in CodeOcean)

Reply:
I'll have a look again

Reply:
Found the bug and fixed it

Reply:
Nice, thank you! I can confirm that the meta data are being exported correctly again.